### PR TITLE
Mongoid 7 Support

### LINKED ISSUE #220: Compatibility with Mongoid 7
[Donate $50 to my charity](https://runwithtfk.org/Profile/PublicPage/61018) and I can take care of it.

### PR DESCRIPTION
**Pull request by dblock**
Closes #220.

Mongoid 6.x to 7 diff: https://github.com/mongodb/mongoid/compare/6.4.0-stable...7.0.0-stable

### CODE REVIEW

**Comment by dblock** [comment]
@jonekdahl  try it?

**Comment by jonekdahl** [comment]
@dblock Our test suite is still failing horribly due to [other bugs](https://jira.mongodb.org/browse/MONGOID-4544) so I can't be sure everything works yet, but it certainly looks like an improvement.

**Comment by dblock** [comment]
Could I please get a code review from maybe @johnnyshields or @aq1018?

**Comment by dblock** [comment]
Found #225 in the process of doing this.

File: Gemfile [hunk 1]
```diff
@@ -2,9 +2,11 @@ source 'https://rubygems.org'
 
 gemspec
 
-case version = ENV['MONGOID_VERSION'] || '~> 6.0.0'
+case version = ENV['MONGOID_VERSION'] || '~> 7.0.0'
 when 'HEAD'
   gem 'mongoid', github: 'mongodb/mongoid'
+when /7/
```
Comment (johnnyshields) [inline Gemfile]: Please use `/\b7\./`   The current regex implies 7 can be anywhere in the string, i.e. 1.7.0.

File: lib/mongoid/history/tracker.rb [hunk 2]
```diff
@@ -179,6 +179,13 @@ def create_standalone
 
       def create_on_parent
         name = association_chain.last['name']
+
+        #
+        # TODO: modifier
+        # klass = trackable_parent.class.relation_class_of(name)
+        # klass.history_trackable_options[:modifier_field] if klass.respond_to?(:history_trackable_options)
+        #
```
Comment (johnnyshields) [inline lib/mongoid/history/tracker.rb]: Please delete this comment, it's just cruft.

**Comment by johnnyshields** [comment]
Please address comments and I'm otherwise fine with this.

File: Gemfile (hunk 1 above)
Comment (dblock) [reply Gemfile]: Good point.

File: lib/mongoid/history/tracker.rb (hunk 2 above)
Comment (dblock) [reply lib/mongoid/history/tracker.rb]: Removed.

**Comment by dblock** [comment]
Updated and green.

**Comment by dblock** [comment]
@johnnyshields you're around? Go for merge. Otherwise let us know and I won't bug you :)

**Comment by dblock** [comment]
Merging/releasing.

**Pull request merged**